src/app/shared/table/table-data-source.ts

```diff
@@ -447,6 +447,14 @@ export abstract class TableDataSource<T extends TableData> {
       this.loadDataImpl().pipe(first()).subscribe((data) => {
         // Set nbr of records
         this.setTotalNumberOfRecords(data.count);
+        // To Remove once the POC is validated
+        if (!Utils.isEmptyArray(data.projectedFields)) {
+          // Display only projected fields headers
+          this.tableColumnsDef.forEach(tableColumnDef => tableColumnDef.visible = data.projectedFields.map(projectedField =>
```

Comment:
I prefer for...of and multiple lines for readability.

Reply:
Done.